### PR TITLE
[MGDOBR-1089] - Cloud events performance testing for RHOSE - 2nd Scenario

### PR DESCRIPTION
**Pull request by afalhambra**
[MGDOBR-1089](https://issues.redhat.com/browse/MGDOBR-1089)

Depends on https://github.com/afalhambra/webhook-perf-test/pull/2

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java [hunk 1]
```diff
@@ -11,20 +13,37 @@ public class EndPointParser {
...
+            return Optional.of(new URL(endpoint).getAuthority());
+        } catch (MalformedURLException e) {
+            return Optional.empty();
+        }
+    }
+
+    public static Optional<Integer> getEndpointPort(String endpoint) {
+        try {
+            int port = new URL(endpoint).getPort();
+            return Optional.of(port != -1 ? port : 80);
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java:43]: Can you please add also handling of default HTTPS port?

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java [hunk 2]
```diff
@@ -11,20 +13,37 @@ public class EndPointParser {
     private EndPointParser() {
     }
 
-    public static Optional<String> getEndpointBaseUrl(String endpoint) {
+    public static Optional<String> getBaseUrl(String endpoint) {
         Matcher matcher = ENDPOINT_URL_REGEX.matcher(endpoint);
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java:17]: One optional idea:
As we have a functionality to parse whole URL now, would it have sense to drop the regex and retrieve all needed info directly from URL object? 
We could drop this whole class and process the URL in `BridgeContext`.
Or event store the URL in `BridgeContext` and construct the needed value in `BridgeEndpointBaseResolver`.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java (hunk 1 above)
Comment (afalhambra) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java]: method not needed, removed.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java (hunk 2 above)
Comment (afalhambra) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java]: agree, thanks for the suggestion.

**Review by sutaakar — APPROVED** [review]
Good job

**Review by jiripetrlik — APPROVED** [review]

**Pull request merged**